festim/hydrogen_transport_problem.py

```diff
@@ -711,3 +713,506 @@ def post_processing(self):
+                object). Defaults to None.
+            petsc_options (dict, optional): petsc options to be passed to the
+                ``festim.NewtonSolver`` object. If None, the default options are:
+                ```
+                default_petsc_options = {
+                    "ksp_type": "preonly",
+                    "pc_type": "lu",
+                    "pc_factor_mat_solver_type": "mumps",
+                }
+                ```
+                Defaults to None.
+        """
+        super().__init__(
+            mesh,
+            subdomains,
+            species,
+            reactions,
+            temperature,
+            sources,
+            initial_conditions,
+            boundary_conditions,
+            settings,
+            exports,
+            traps,
+        )
+        self.interfaces = interfaces or []
+        self.surface_to_volume = surface_to_volume or {}
+        default_petsc_options = {
+            "ksp_type": "preonly",
+            "pc_type": "lu",
+            "pc_factor_mat_solver_type": "mumps",
+        }
+        self.petsc_options = petsc_options or default_petsc_options
+
+    def initialise(self):
+        # check that all species have a list of F.VolumeSubdomain as this is
+        # different from F.HydrogenTransportProblem
+        for spe in self.species:
+            if not isinstance(spe.subdomains, list):
+                raise TypeError("subdomains attribute should be list")
+
+        self.define_meshtags_and_measures()
+
+        # create submeshes and transfer meshtags to subdomains
+        for subdomain in self.volume_subdomains:
+            subdomain.create_subdomain(self.mesh.mesh, self.volume_meshtags)
+            subdomain.transfer_meshtag(self.mesh.mesh, self.facet_meshtags)
+
+        for interface in self.interfaces:
+            interface.mt = self.volume_meshtags
+            interface.parent_mesh = self.mesh.mesh
+
+        self.create_species_from_traps()
+
+        self.t = fem.Constant(self.mesh.mesh, 0.0)
+        if self.settings.transient:
+            # TODO should raise error if no stepsize is provided
+            # TODO Should this be an attribute of festim.Stepsize?
+            self.dt = F.as_fenics_constant(
+                self.settings.stepsize.initial_value, self.mesh.mesh
+            )
+
+        self.define_temperature()
+        self.create_source_values_fenics()
+        self.create_flux_values_fenics()
+        self.create_initial_conditions()
+
+        for subdomain in self.volume_subdomains:
+            self.define_function_spaces(subdomain)
+            ct_r = dolfinx.mesh.meshtags(
+                self.mesh.mesh,
+                self.mesh.mesh.topology.dim,
+                subdomain.submesh_to_mesh,
+                np.full_like(subdomain.submesh_to_mesh, 1, dtype=np.int32),
+            )
+            subdomain.dx = ufl.Measure(
+                "dx", domain=self.mesh.mesh, subdomain_data=ct_r, subdomain_id=1
+            )
```

Comment:
```suggestion
            subdomain_dx = self.parent_dx(subdomain_marker)
```
If we define a `parent_dx` in the initialization of the TransportProblem.
We sit down and fix this later today.